src/mpi/comm/comm_impl.c

```diff
@@ -587,23 +626,46 @@ int MPIR_Intercomm_create_from_groups_impl(MPIR_Group * local_group_ptr, int loc
 {
     int mpi_errno = MPI_SUCCESS;
 
-    MPIR_Assert(MPIR_Process.comm_world);
+    MPIR_Session *session = local_group_ptr->session_ptr;
+    MPIR_ERR_CHKANDJUMP(session != remote_group_ptr->session_ptr, mpi_errno, MPI_ERR_OTHER,
+                        "**session_mixed");
```

Comment:
not sure I understand that error condition. what does it mean for sessions to be mixed if a session is a local only resource?

Reply:
OK, different sessions in the same process. I think I got it now.